llvm/lib/Target/PowerPC/MCTargetDesc/PPCMCTargetDesc.cpp

```diff
@@ -423,7 +427,8 @@ class PPCTargetXCOFFStreamer : public PPCTargetStreamer {
   }
 
   void emitMachine(StringRef CPU) override {
-    llvm_unreachable("Machine pseudo-ops are invalid for XCOFF.");
+    MCXCOFFStreamer &XCOFFStreamer = static_cast<MCXCOFFStreamer &>(Streamer);
+    XCOFFStreamer.getAssembler().getWriter().setCPU(CPU);
```

Comment:
With the other changes this would become:
```suggestion
    static_cast<XCOFFObjectWriter>(Streamer.getAssembler().getWriter()).setCPU(CPU);
```